src/components/views/ViewDataTable/utils.tsx

```diff
@@ -100,3 +100,49 @@ export function makeGridColDef(
 
   return colDef;
 }
+
+export const viewQuickSearch = (
```

Comment:
I think that this solution might be unnecessarily complex. Is it not possible to just use the `valueGetter` approach already implemented for all columns, and used for sorting and filtering?

If it's not, I think the implementation here is still very complex in that it takes a very general-purpose solution to what is not a general-purpose problem, but still rebuilds the "index" (joined strings) every time the search query updates.

If using `valueGetter` is not possible, I still think this code can be either simplified, or optimized, and I prefer simplified as long as we don't have actual performance problems.

Simplification to me would include just looping over the rows and for each row making a comparison against each relevant string (which requires column type-aware handling of each column).

Reply:
Here's a suggestion on how to do what I'm proposing above. I've written this to illustrate, and have not tested it.

```tsx
export const viewQuickSearch = (
  rows: ZetkinViewRow[],
  columns: ZetkinViewColumn[],
  quickSearch: string
): ZetkinViewRow[] => {
  if (!quickSearch) {
    return rows;
  }

  return rows.filter((row) => {
    // For each row, go through columns and check whether any
    // of them match the search string. Stop at first match.
    return columns.some((column, colIdx) => {
      const value = row.content[colIdx];

      if (!value) {
        return false;
      }

      if (column.type == COLUMN_TYPE.PERSON_FIELD) {
        // Basic string comparison for PERSON_FIELD cells
        const valueString = value as string;

        if (valueString.includes(quickSearch)) {
          return true;
        }
      } else if (
        column.type == COLUMN_TYPE.PERSON_NOTES ||
        column.type == COLUMN_TYPE.SURVEY_SUBMITTED
      ) {
        // For PERSON_NOTES and SURVEY_SUBMISSION, the
        // value will be an array of objects. Look for their
        // .text property to match.
        const valueArray = value as [{ text: string }];
        const doesMatch = valueArray.some((obj) =>
          obj.text.includes(quickSearch)
        );

        if (doesMatch) {
          return true;
        }
      }

      return false;
    });
  });
};
```